Maven problems on a clean room environment

```
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[ERROR] Resolving expression: '${final.name}': Detected the following recursive expression cycle in 'final.name': [final.name] @ 
[ERROR] Resolving expression: '${final.name}': Detected the following recursive expression cycle in 'final.name': [final.name] @ 
[ERROR] 'dependencies.dependency.version' for io.quarkus:quarkus-vertx-web:jar is missing. @ line 49, column 21
 @ 
[ERROR] The build could not read 2 projects -> [Help 1]
[ERROR]   
[ERROR]   The project org.graalvm.tests.integration:quarkus-json:1.0.0-SNAPSHOT (/home/tester/karm/mandrel-integration-tests/apps/quarkus-json/pom.xml) has 2 errors
[ERROR]     Resolving expression: '${final.name}': Detected the following recursive expression cycle in 'final.name': [final.name] -> [Help 2]
[ERROR]     Resolving expression: '${final.name}': Detected the following recursive expression cycle in 'final.name': [final.name] -> [Help 2]
[ERROR]   
[ERROR]   The project org.acme:quarkus-vertx:1.0.0-SNAPSHOT (/home/tester/karm/mandrel-integration-tests/apps/quarkus-vertx/pom.xml) has 1 error
[ERROR]     'dependencies.dependency.version' for io.quarkus:quarkus-vertx-web:jar is missing. @ line 49, column 21

```


```
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Maven home: /opt/apache-maven-3.6.3
Java version: 17.0.8.1, vendor: Eclipse Adoptium, runtime: /opt/jdk-17.0.8.1+1
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "5.14.0-319.el9.aarch64", arch: "aarch64", family: "unix"

```

--- a/apps/quarkus-json/pom.xml
+++ b/apps/quarkus-json/pom.xml
@@ -84,5 +84,27 @@
                 <quarkus.package.type>native</quarkus.package.type>
             </properties>
         </profile>
+        <profile>
+            <id>custom-final-name-not-set</id>
+            <activation>
+                <property>
+                    <name>!custom.final.name</name>
+                </property>
+            </activation>
+            <properties>
+                <final.name>quarkus-json-test</final.name>
+            </properties>
+        </profile>
+        <profile>
+            <id>custom-final-name-set</id>
+            <activation>
+                <property>
+                    <name>custom.final.name</name>
+                </property>
+            </activation>
+            <properties>
+                <final.name>${custom.final.name}</final.name>
+            </properties>
+        </profile>
     </profiles>
 </project>

--- a/apps/quarkus-vertx/pom.xml
+++ b/apps/quarkus-vertx/pom.xml
@@ -48,7 +48,7 @@
         </dependency>
         <dependency>
             <groupId>io.quarkus</groupId>
-            <artifactId>quarkus-vertx-web</artifactId>
+            <artifactId>quarkus-reactive-routes</artifactId>
         </dependency>
         <dependency>
             <groupId>io.quarkus</groupId>

--- a/apps/quarkus-vertx/quarkus_3.x.patch
+++ b/apps/quarkus-vertx/quarkus_3.x.patch
@@ -1,16 +1,3 @@
-diff --git a/apps/quarkus-vertx/pom.xml b/apps/quarkus-vertx/pom.xml
-index 4bbe1d3..7d47c9b 100644
---- a/apps/quarkus-vertx/pom.xml
-+++ b/apps/quarkus-vertx/pom.xml
-@@ -48,7 +48,7 @@
-         </dependency>
-         <dependency>
-             <groupId>io.quarkus</groupId>
--            <artifactId>quarkus-vertx-web</artifactId>
-+            <artifactId>quarkus-reactive-routes</artifactId>
-         </dependency>
-         <dependency>
-             <groupId>io.quarkus</groupId>
 diff --git a/apps/quarkus-vertx/src/main/java/org/acme/vertx/FruitResource.java b/apps/quarkus-vertx/src/main/java/org/acme/vertx/FruitResource.java
 index cf0796c..66c53c2 100644
 --- a/apps/quarkus-vertx/src/main/java/org/acme/vertx/FruitResource.java

--- a/testsuite/src/it/java/org/graalvm/tests/integration/utils/BuildAndRunCmds.java
+++ b/testsuite/src/it/java/org/graalvm/tests/integration/utils/BuildAndRunCmds.java
@@ -94,14 +94,14 @@ public enum BuildAndRunCmds {
                             "-R:MaxHeapSize=" + MX_HEAP_MB + "m," +
                             "-H:-ParseOnce" +
                             GRAALVM_BUILD_OUTPUT_JSON_FILE + "-ParseOnce",
-                    "-Dfinal.name=quarkus-json_-ParseOnce"},
+                    "-Dcustom.final.name=quarkus-json_-ParseOnce"},
             new String[]{"mvn", "package", "-Pnative", "-Dquarkus.version=" + QUARKUS_VERSION.getVersionString(),
                     "-Dquarkus.native.additional-build-args=" +
                             "-R:MaxHeapSize=" + MX_HEAP_MB + "m," +
                             "-H:+ParseOnce" +
                             GRAALVM_BUILD_OUTPUT_JSON_FILE + "+ParseOnce",
-                    "-Dfinal.name=quarkus-json_+ParseOnce"},
-            new String[]{"mvn", "package", "-Dfinal.name=quarkus-json", "-Dquarkus.version=" + QUARKUS_VERSION.getVersionString()},
+                    "-Dcustom.final.name=quarkus-json_+ParseOnce"},
+            new String[]{"mvn", "package", "-Dcustom.final.name=quarkus-json", "-Dquarkus.version=" + QUARKUS_VERSION.getVersionString()},
             //-XX:+UseShenandoahGC
             //-XX:+UseSerialGC
             //-XX:+UseG1GC
@@ -115,8 +115,8 @@ public enum BuildAndRunCmds {
                     "-Dquarkus.native.additional-build-args=" +
                             "-R:MaxHeapSize=" + MX_HEAP_MB + "m" +
                             GRAALVM_BUILD_OUTPUT_JSON_FILE,
-                    "-Dfinal.name=quarkus-json"},
-            new String[]{"mvn", "package", "-Dfinal.name=quarkus-json", "-Dquarkus.version=" + QUARKUS_VERSION.getVersionString()},
+                    "-Dcustom.final.name=quarkus-json"},
+            new String[]{"mvn", "package", "-Dcustom.final.name=quarkus-json", "-Dquarkus.version=" + QUARKUS_VERSION.getVersionString()},
             new String[]{"perf", "stat", "--delay", "2000", "java", "-Xlog:gc", "-XX:+UseSerialGC", "-Xmx" + MX_HEAP_MB + "m", "-jar", "target/quarkus-app/quarkus-run.jar"},
             new String[]{"perf", "stat", "--delay", "1000", "./target/quarkus-json-runner", "-XX:+PrintGC"},
     }),

--- a/testsuite/src/it/java/org/graalvm/tests/integration/utils/Commands.java
+++ b/testsuite/src/it/java/org/graalvm/tests/integration/utils/Commands.java
@@ -1062,7 +1062,7 @@ public class Commands {
      *               "-R:MaxHeapSize=" + MX_HEAP_MB + "m," +
      *               "-H:-ParseOnce," +
      *               "-H:BuildOutputJSONFile=quarkus-json_minus-ParseOnce.json",
-     *               "-Dfinal.name=quarkus-json_-ParseOnce"},
+     *               "-Dcustom.final.name=quarkus-json_-ParseOnce"},
      * //@formatter:on
      */
     public static void builderRoutine(int steps, Apps app, StringBuilder report, String cn, String mn, File appDir,

